pkg/scheduler/preemption/preemption.go

```diff
@@ -144,6 +144,9 @@ func (p *Preemptor) getTargets(preemptionCtx *preemptionCtx) []*Target {
 		return nil
 	}
 	sort.Slice(candidates, candidatesOrdering(candidates, preemptionCtx.preemptorCQ.Name, p.clock.Now()))
+	if p.enableFairSharing {
```

Comment:
Hm, actually, is it guaranteed to generate the same targets? Probably not, but hard to say cause we don't test much this branch when fair sharing enabled. Can you justify the new logic is better in this case?

Reply:
The justification is in the PR description - I argue that the code base is easier to reason about/debug when we always use the FairSharing function when FairSharing is enabled. Otherwise, need to reason about the other path being taken

Reply:
What about the first question: " actually, is it guaranteed to generate the same targets? "?

Reply:
Basically, I'm not so comfortable calling a cleanup something which has user-facing consequences, even if it makes the code cleaner. This is why we have the "Does this PR introduce a user-facing change?" point in the description.

Reply:
I expect it to return the same result. They process candidates in the same order and have identical logic for same CQ (minimalPreemptions is called with allowBorrowing=true)

https://github.com/kubernetes-sigs/kueue/blob/69e1f2ca05de4892ae94b4042db199cb3ef20c09/pkg/scheduler/preemption/preemption.go#L307-L315

https://github.com/kubernetes-sigs/kueue/blob/69e1f2ca05de4892ae94b4042db199cb3ef20c09/pkg/scheduler/preemption/preemption.go#L392-L399